assets/javascripts/discourse/components/ai-tool-editor-form.gjs

```diff
@@ -0,0 +1,359 @@
+              </collection.Field>
+            </row.Col>
+
+            {{#if collectionData.enum}}
+              <row.Col @size={{8}}>
+                <collection.Collection @name="enumData" as |child childIndex|>
+                  <form.Container class="ai-tool-parameter__enum-values">
+                    <child.Field
+                      @name="enumValue"
+                      @title={{i18n "discourse_ai.tools.enum_value"}}
+                      as |field|
+                    >
+                      <field.Input />
+                    </child.Field>
+
+                    {{#if (gt collectionData.enumData.length 1)}}
+                      <form.Button
+                        class="btn-danger"
+                        @icon="trash-can"
+                        @action={{fn child.remove childIndex}}
+                      />
+                    {{/if}}
+
+                    <form.Button
+                      @icon="plus"
+                      @label="discourse_ai.tools.add_enum_value"
+                      @action={{fn
+                        form.addItemToCollection
+                        "enumData"
+                        (array "")
+                      }}
+                    />
+                  </form.Container>
+
+                </collection.Collection>
+              </row.Col>
+            {{/if}}
+          </form.Row>
+          <form.Row as |row|>
+            <row.Col class="ai-tool-parameter-actions">
+              <form.Button
+                @label="discourse_ai.tools.remove_parameter"
+                @icon="trash-can"
+                @action={{fn collection.remove index}}
+                class="btn-danger"
+              />
+            </row.Col>
+          </form.Row>
+        </form.Container>
+      </form.Collection>
+
+      <form.Button
+        @icon="plus"
+        @label="discourse_ai.tools.add_parameter"
+        @action={{fn
+          form.addItemToCollection
+          "parameters"
+          (hash name="" type="string" description="" required=false enum=false)
+        }}
+      />
+
+      {{! SCRIPT }}
+      <form.Field
+        @name="script"
+        @title={{i18n "discourse_ai.tools.script"}}
+        @validation="required|length:1,100000"
+        @format="full"
+        as |field|
+      >
+        <field.Code @lang="javascript" @height={{600}} />
+      </form.Field>
+
+      {{! UPLOADS }}
+      {{#if this.siteSettings.ai_embeddings_enabled}}
+        <form.Field
+          @name="rag_uploads"
+          @title={{i18n "discourse_ai.rag.uploads.title"}}
+          @tooltip={{this.ragUploadsDescription}}
+          as |field|
+        >
+          <field.Custom>
+            <RagUploader
+              @target={{@editingModel}}
+              @updateUploads={{fn this.updateUploads form.addItemToCollection}}
+              @onRemove={{fn this.removeUpload form}}
+              @allowImages={{@settings.rag_images_enabled}}
+            />
+            <RagOptions
```

Comment:
@keegangeorge - I created an FK version of this component called `RagOptionsFK`, can we combine them now that this form uses form kit too?